### PR TITLE
Fix removing staff on publish

### PR DESCRIPTION
**Pull request by asadazam93**
### Description
Fix the publish button to remove existing instructors.

**Sandbox**
N/A

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [ ] @attiyaIshaque      
- [ ] @awaisdar001

### Post-review
- [ ] Rebase and squash commits

### CODE REVIEW

File: course_discovery/apps/publisher/api/v1/views.py [hunk 1]
```diff
@@ -154,6 +154,7 @@ def publish_to_discovery(self, partner, course_run):
             defaults=defaults
         )
         discovery_course_run.transcript_languages.add(*course_run.transcript_languages.all())
+        discovery_course_run.staff.clear()
```
Comment (awaisdar001) [inline course_discovery/apps/publisher/api/v1/views.py:157]: This looks straight forwared, but can you add a relevent test for this scenrio?

Comment (asadazam93) [reply course_discovery/apps/publisher/api/v1/views.py:157]: Test case added

File: course_discovery/apps/publisher/api/v1/tests/test_views.py [hunk 2]
```diff
@@ -235,6 +235,34 @@ def test_publish_seat_without_upgrade_deadline(self, mock_access_token):
...
+        """
+        publisher_course_run = self._create_course_run_for_publication()
+
+        partner = publisher_course_run.course.organizations.first().partner
+        self._set_test_client_domain_and_login(partner)
+
+        self._mock_studio_api_success(publisher_course_run)
+        self._mock_ecommerce_api(publisher_course_run)
+
+        url = reverse('publisher:api:v1:course_run-publish', kwargs={'pk': publisher_course_run.pk})
```
Comment (awaisdar001) [inline course_discovery/apps/publisher/api/v1/tests/test_views.py]: `url` => `publish_url`

File: course_discovery/apps/publisher/api/v1/tests/test_views.py [hunk 3]
```diff
@@ -235,6 +235,34 @@ def test_publish_seat_without_upgrade_deadline(self, mock_access_token):
...
+        response = self.client.post(url, {})
+        assert response.status_code == 200
+        discovery_course_run = CourseRun.objects.get(key=publisher_course_run.lms_course_id)
+        assert discovery_course_run.staff.all().count() == 2
+
+        publisher_course_run.staff.clear()
+        publisher_course_run.staff = PersonFactory.create_batch(1)
```
Comment (awaisdar001) [inline course_discovery/apps/publisher/api/v1/tests/test_views.py:259]: you have not saved the `publisher_course_run` after assigning it new (1) staff?

File: course_discovery/apps/publisher/api/v1/tests/test_views.py [hunk 4]
```diff
@@ -235,6 +235,34 @@ def test_publish_seat_without_upgrade_deadline(self, mock_access_token):
...
+        url = reverse('publisher:api:v1:course_run-publish', kwargs={'pk': publisher_course_run.pk})
```
Comment (awaisdar001) [inline course_discovery/apps/publisher/api/v1/tests/test_views.py]: I believe this is not required as you have the same url already.

File: course_discovery/apps/publisher/api/v1/tests/test_views.py (hunk 3 above)
Comment (asadazam93) [reply course_discovery/apps/publisher/api/v1/tests/test_views.py:259]: There is no need to save in this case. Assigning a ManyToMany field automatically saves the model instance.

**Review by awaisdar001 — APPROVED** [review]

**Review by attiyaIshaque — APPROVED** [review]
Once the build green.  :+1

**Pull request merged**